File: src/main/java/FlyAway/reservation/dto/ReservationDetailsEmployeeDto.java
package FlyAway.reservation.dto;

import FlyAway.client.dto.ClientDto;
import FlyAway.flight.aircraft.CabinClass;
import FlyAway.flight.dto.FlightDetailsEmployeeDto;
import FlyAway.reservation.ReservationStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

public record ReservationDetailsEmployeeDto(
        UUID id,
        ReservationStatus status,
        ClientDto clientDto,
        CabinClass cabinClass,
        Integer seatNumber,
        BigDecimal price,
        LocalDateTime reservationDate,
        FlightDetailsEmployeeDto flightDto
) {
}

File: frontend/src/app/services/models/reservation-details-employee-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ClientDto } from '../models/client-dto';
import { FlightDetailsEmployeeDto } from '../models/flight-details-employee-dto';
export interface ReservationDetailsEmployeeDto {
  cabinClass?: 'FIRST' | 'BUSINESS' | 'ECONOMY';
  clientDto?: ClientDto;
  flightDto?: FlightDetailsEmployeeDto;
  id?: string;
  price?: number;
  reservationDate?: string;
  seatNumber?: number;
  status?: 'PENDING' | 'ACTIVE' | 'CANCELLED' | 'EXPIRED' | 'COMPLETED' | 'FAILED';
}
